### DISCUSSION TITLE
Ruleset for committer email, when there is none

### DISCUSSION THREAD

**Discussion by sandstrom** [community/community#127396] [Repositories] [answered]
### Select Topic Area

Question

### Body

Many commits will only have an author, and no committer.

We want to enforce a regex pattern (specific email domain) for the committer email, but only if there is one.

My initial attempt was `^[a-z]+@example\.com$`, and then I've tried `(^[a-z]+@example\.com$|^$)`. None of them works.

Trying to set this up, but cannot figure out how to make it work.

Getting the error `Committer email must match a given regex pattern`.

<img width="754" alt="image" src="https://github.com/community/community/assets/122287/88a774f3-f552-4ccc-88d7-462393c0398f">

**Comment by sandstrom** [comment]
@patrick-knight happy to hear your thoughts on this one 😄

**Reply by queenofcorgis** [reply]
Hi there,

Please note, contacting any member of GitHub Staff via unsolicited mentions or pings is strongly discouraged and may be considered a violation of our prohibition against harassment. Please review our Code of Conduct for more information.

Let's work together to keep GitHub Community a place where people feel safe to participate by being respectful of them and their time.

**Reply by sandstrom** [reply]
Wow, harassment? 

I was just asking because he has commented on some issues related to rulesets on this forum, and seem to be a PM working on it. He is free ignore me if he doesn't want to answer [same for anyone else working at Github].

**Reply by dserodio** [reply]
It's a general rule in _Internet forums_ (and on IRC before the existence of "Internet forums") that sending unsolicited DMs is frowned upon.
I'm sure that GH Staff already have an unmanageable amount of notifications

**Reply by sandstrom** [reply]
@dserodio Look, this is the public feedback forum for the for-profit company Github. 

If someone working as a PM (where one of the assigned work duties often is to take in feedback from customers), feels harassed by me asking a question about said product in Github's public forum for feedback and discussions, then I'm sorry!

If Github staff feels that it's a recurring problem that customers are communicating with them through their public forum -- and that they feel harassed by it -- then I'd suggest they either ban their customers from it, or ask a Github developer to silence all the @-mentions from their public forum.

***

> I'm sure that GH Staff already have an unmanageable amount of notifications

I've had great discussions with several other Github PMs and engineers on this forum. I'd assume they participate here because they overall feel that it adds some value to them. So far none has said that they feel harassed by it.

**Comment by patrick-knight** [comment, marked as answer]
Right now we fail immediately if there is no metadata associated with a specific commit property, but I'm curious to understand the scenarios you have where there is an author but no committer info.

**Reply by sandstrom** [reply]
Thanks for chiming in, much appreciated! 😄 Also, of course (related to above), if you feel "harassed" by me using the @-feature let me know, and I'll never ping you again!

We can chalk this one up to operator error 😅 I didn't realize the  Git log didn't show committer without `--format=full`. So I thought it was an optional value, set only on commits when the committer is different from the author (hence my attempt below with matching an empty string).

But that's not the case. So I think my non-matching pattern is because the merge commits will be with committer `Github <noreply@github.com>`. I've updated my ruleset and will sett if this solves it!

Two thoughts on this:

1. Most of this is my mistake, but perhaps it would help if the ruleset insight contained information about the text that the pattern didn't match. Obviously more data for you to store though, so you'd have to decide if that makes sense. But would certainly help with troubleshooting.
2. "Right now we fail immediately if there is no metadata associated with a specific commit property" sounds reasonable given how the Git data is structured. I wouldn't change it.

***

#### Off topic

While I have your ear: I don't know if this is your area or another team, but semi-linear merge support (in general, and for the merge queue) would be a great addition to Github.

More details:

- https://github.com/orgs/community/discussions/14863
- https://github.com/orgs/community/discussions/8940

**Reply by patrick-knight** [reply]
Agreed, we need to drive clarity in the UI to make sure your rule works right the first time without manually testing. We're thinking about a few ways to do that well inclusive of more links to docs in product.

I'll pass on the semi-linear feedback to the right folks.

**Reply by sandstrom** [reply]
Thanks @patrick-knight, much appreciated!

**Reply by yermulnik** [reply]
I'm working on the same task as topicstarter did. Am I getting this right: given the legit commit (as in NOT those that have no metadata associated) committer email is either committer author email (or someone else's from the Org who actually pushed commit) or `noreply@github.com` (for merges and similar actions)? So I should be good to set committer email regex to `^([^@]+@example\.com|noreply@github\.com)$` and committer author email to `^[^@]+@example\.com$`, right?
Thanks in advance.

**Reply by sandstrom** [reply]
@yermulnik I don't work at Github so cannot confirm, but my understanding is that's how it works (i.e. those regexes should work).

**Reply by yermulnik** [reply]
Thanks @sandstrom and apologies for the confusion: I meant to address my question to @patrick-knight

**Comment by jamesrcutmore** [comment]
Hi guys, where do I ask a question in this community, I need help deleting repo. Thanks

**Reply by sandstrom** [reply]
There are docs here:

https://docs.github.com/en/repositories/creating-and-managing-repositories/deleting-a-repository
